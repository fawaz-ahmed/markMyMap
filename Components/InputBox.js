import React, { forwardRef, memo } from 'react';
import { TextField } from 'react-native-material-textfield';
import Colors from '../Colors';

export default memo(forwardRef((props, ref) => (
  <TextField
    ref={ref}
    {...props}
    textColor={Colors.inputTextColor}
    fontSize={16}
    titleFontSize={12}
    labelFontSize={12}
    labelPadding={12}
    inputContainerPadding={8}
    activeLineWidth={0.5}
    tintColor={Colors.inputTintColor}
    baseColor={Colors.inputTintColor}
    containerStyle={{
      marginBottom: props.marginBottom || 0,
    }}
  />
)));
