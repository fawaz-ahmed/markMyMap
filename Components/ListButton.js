import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, View } from 'react-native';
import styles from '../Styles/Button';

export default class ListButton extends PureComponent {

  render() {
    const { onPress } = this.props;

    return (
      <TouchableOpacity onPress={onPress} style={styles.listContainer}>
        <View style={styles.buttonContainer}>
          <Text style={styles.buttonText}>{'â‰¡'}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

ListButton.propTypes = {
  onPress: PropTypes.func,
};
