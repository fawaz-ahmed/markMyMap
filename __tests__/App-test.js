import 'react-native';
import React from 'react';
// import renderer from 'react-test-renderer';
import Enzyme, { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AddButton from '../Components/AddButton';
import ListButton from '../Components/ListButton';
import InputBox from '../Components/InputBox';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing AddButton', () => {
  const wrapper = shallow(<AddButton />);
  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Test onPress', () => {
    let isPressed = false;
    const mockFn = () => { isPressed = true; }
    wrapper.setProps({ onPress: mockFn });
    wrapper.simulate('press');
    expect(isPressed).toBeTruthy();
  });
});


describe('Testing ListButton', () => {
  const wrapper = shallow(<ListButton />);
  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Test onPress', () => {
    let isPressed = false;
    const mockFn = () => { isPressed = true; }
    wrapper.setProps({ onPress: mockFn });
    wrapper.simulate('press');
    expect(isPressed).toBeTruthy();
  });
});

describe('Testing InputBox', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<InputBox label={'label'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Test value', () => {
    const testString = 'this is a test string';
    const instanceOf = render(<InputBox value={testString} />);
    expect(instanceOf.toString().indexOf(testString) !== -1).toBeTruthy();
  });
});
