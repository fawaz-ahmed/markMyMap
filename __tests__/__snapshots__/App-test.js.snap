// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddButton renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "bottom": 35,
      "position": "absolute",
      "right": 23,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0099CC",
        "borderRadius": 23.5,
        "height": 47,
        "justifyContent": "center",
        "paddingBottom": 3,
        "width": 47,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
        }
      }
    >
      +
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Testing InputBox renders as expected 1`] = `
<TextField
  activeLineWidth={0.5}
  animationDuration={225}
  autoCapitalize="sentences"
  baseColor="rgb(135,135,135)"
  containerStyle={
    Object {
      "marginBottom": 0,
    }
  }
  disableFullscreenUI={true}
  disabled={false}
  disabledLineType="dotted"
  disabledLineWidth={1}
  editable={true}
  errorColor="rgb(213, 0, 0)"
  fontSize={16}
  inputContainerPadding={8}
  label="label"
  labelFontSize={12}
  labelHeight={32}
  labelPadding={12}
  lineWidth={0.5}
  textColor="rgb(51,51,51)"
  tintColor="rgb(135,135,135)"
  titleFontSize={12}
  underlineColorAndroid="transparent"
/>
`;

exports[`Testing ListButton renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "bottom": 92,
      "position": "absolute",
      "right": 23,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0099CC",
        "borderRadius": 23.5,
        "height": 47,
        "justifyContent": "center",
        "paddingBottom": 3,
        "width": 47,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
        }
      }
    >
      â‰¡
    </Text>
  </View>
</TouchableOpacity>
`;
